pyoffers/models/tags.py

```diff
@@ -0,0 +1,98 @@
+
+    def add_to_advertiser(self, advertiser_id):
+        return self._manager.add_to_advertiser(self.id, advertiser_id=advertiser_id)
+
+    def remove_from_offer(self, offer_id):
+        return self._manager.remove_from_offer(self.id, offer_id=offer_id)
+
+    def remove_from_affiliate(self, affiliate_id):
+        return self._manager.remove_from_affiliate(self.id, affiliate_id=affiliate_id)
+
+    def remove_from_advertiser(self, advertiser_id):
+        return self._manager.remove_from_advertiser(self.id, advertiser_id=advertiser_id)
+
+
+class TagManager(ModelManager):
+    model = Tag
+    name = 'tags'
+    generic_methods = (
+        'find_by_id',
+        'find_all',
+        'delete',
+        'update',
+        'create',
+    )
+
+    def find_all_offer_tag_relations(self, **kwargs):
+        return self._call('findAllOfferTagRelations', target_class='TagRelation', single_result=False, **kwargs)
+
+    def find_all_affiliate_tag_relations(self, **kwargs):
+        return self._call('findAllAffiliateTagRelations', target_class='TagRelation', single_result=False, **kwargs)
+
+    def find_all_advertiser_tag_relations(self, **kwargs):
+        return self._call('findAllAdvertiserTagRelations', target_class='TagRelation', single_result=False, **kwargs)
+
+    def add_to_offer(self, tag_id, offer_id):
+        return self._call('addToOffer', target_class='TagRelation', tag_id=tag_id, offer_id=offer_id)
+
+    def add_to_affiliate(self, tag_id, affiliate_id):
+        return self._call('addToAffiliate', target_class='TagRelation', tag_id=tag_id, affiliate_id=affiliate_id)
+
+    def add_to_advertiser(self, tag_id, advertiser_id):
+        return self._call('addToAdvertiser', target_class='TagRelation', tag_id=tag_id, advertiser_id=advertiser_id)
+
+    def set_for_offer(self, tag_ids, offer_id):
+        return self._call('setForOffer', target_class='TagRelation', tag_ids=tag_ids, offer_id=offer_id)
+
+    def set_for_affiliate(self, tag_ids, affiliate_id):
+        return self._call('setForAffiliate', target_class='TagRelation', tag_ids=tag_ids, affiliate_id=affiliate_id)
+
+    def set_for_advertiser(self, tag_ids, advertiser_id):
+        return self._call('setForAdvertiser', target_class='TagRelation', tag_ids=tag_ids, advertiser_id=advertiser_id)
+
+    def remove_from_offer(self, tag_id, offer_id):
+        return self._call('removeFromOffer', tag_id=tag_id, offer_id=offer_id)
+
+    def remove_from_affiliate(self, tag_id, affiliate_id):
+        return self._call('removeFromAffiliate', tag_id=tag_id, affiliate_id=affiliate_id)
+
+    def remove_from_advertiser(self, tag_id, advertiser_id):
+        return self._call('removeFromAdvertiser', tag_id=tag_id, advertiser_id=advertiser_id)
+
+    def remove_from_advertiser_by_relational_id(self, id):
+        return self._call('removeFromAdvertiserByRelationalId', id=id)
+
+    # todo: uncomment when HO fixes bugs in this method
```

Comment:
Please, don't add commented code. Either add it or not.
I'd suggest to add it

Reply:
Added. Marked corresponding tests as skipped.